### ISSUE TITLE
hyper 0.14.27 ICE w/ RUSTFLAGS=" -Z threads=8"

### ISSUE THREAD

**Issue by schell** [rust-lang/rust#117768]
You should be able to recreate this by compiling `hyper = "0.14.27"` with `RUSTFLAGS="-Z threads=8`.

### Error output

```
thread '<unnamed>' panicked at compiler/rustc_query_system/src/query/job.rs:543:9:
deadlock detected
stack backtrace:
   0:        0x103b0f180 - std::backtrace::Backtrace::create::hbce9c5ad34460b69
   1:        0x10cbe4c40 - <alloc[ce5db3987350ea9e]::boxed::Box<rustc_driver_impl[c1f29ae1717eb959]::install_ice_hook::{closure#0}> as core[e17b1bfd351ee445]::ops::function::Fn<(&dyn for<'a, 'b> core[e17b1bfd351ee445]::ops::function::Fn<(&'a core[e17b1bfd351ee445]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[e17b1bfd351ee445]::marker::Send + core[e17b1bfd351ee445]::marker::Sync, &core[e17b1bfd351ee445]::panic::panic_info::PanicInfo)>>::call
   2:        0x103b2711c - std::panicking::rust_panic_with_hook::h507f48f3262f9677
   3:        0x103b26ec0 - std::panicking::begin_panic_handler::{{closure}}::hdafcf7d3ee3805fb
   4:        0x103b246d0 - std::sys_common::backtrace::__rust_end_short_backtrace::hed5651d8ca22bdf3
   5:        0x103b26c80 - _rust_begin_unwind
   6:        0x103b9df64 - core::panicking::panic_fmt::h5987b735963c557e
   7:        0x10df3be00 - rustc_query_system[f767e0e86444d493]::query::job::deadlock
   8:        0x10cbc6714 - std[9107502e9e20a56b]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[fdf73c8bbfbab0ed]::util::run_in_thread_pool_with_globals<rustc_interface[fdf73c8bbfbab0ed]::interface::run_compiler<core[e17b1bfd351ee445]::result::Result<(), rustc_span[a04965e70f1136e6]::ErrorGuaranteed>, rustc_driver_impl[c1f29ae1717eb959]::run_compiler::{closure#1}>::{closure#0}, core[e17b1bfd351ee445]::result::Result<(), rustc_span[a04965e70f1136e6]::ErrorGuaranteed>>::{closure#2}::{closure#1}, ()>
   9:        0x10cbcacf4 - <<std[9107502e9e20a56b]::thread::Builder>::spawn_unchecked_<rustc_interface[fdf73c8bbfbab0ed]::util::run_in_thread_pool_with_globals<rustc_interface[fdf73c8bbfbab0ed]::interface::run_compiler<core[e17b1bfd351ee445]::result::Result<(), rustc_span[a04965e70f1136e6]::ErrorGuaranteed>, rustc_driver_impl[c1f29ae1717eb959]::run_compiler::{closure#1}>::{closure#0}, core[e17b1bfd351ee445]::result::Result<(), rustc_span[a04965e70f1136e6]::ErrorGuaranteed>>::{closure#2}::{closure#1}, ()>::{closure#1} as core[e17b1bfd351ee445]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  10:        0x103b3014c - std::sys::unix::thread::Thread::new::thread_start::h839aee321f78783f
  11:        0x19bc77fa8 - __pthread_joiner_wake


rustc version: 1.75.0-nightly (fdaaaf9f9 2023-11-08)
platform: aarch64-apple-darwin
```

**Comment by SparrowLii** [comment]
Can you reproduce this bug using the latest nightly version to get the new error output?

**Comment by schell** [comment]
After updating to the latest nightly and running the same thing (with a clean project) I immediately get this error:
[output.txt](https://github.com/rust-lang/rust/files/13593144/output.txt)
[rustc-ice-2023-12-07T00_29_40-53983.txt](https://github.com/rust-lang/rust/files/13593151/rustc-ice-2023-12-07T00_29_40-53983.txt)

**Comment by SparrowLii** [comment]
It would be fixed by #118589

**Comment by SparrowLii** [comment]
close this with #118709
